clarity/contracts/arkadiko-swap-v1-1.clar

```diff
@@ -242,17 +355,25 @@
         name: pair-name,
       })
     )
+    (asserts! (is-eq contract-caller (contract-call? .arkadiko-dao get-dao-owner)) (err ERR-NOT-AUTHORIZED))
     (asserts!
       (and
         (is-none (map-get? pairs-data-map { token-x: token-x, token-y: token-y }))
         (is-none (map-get? pairs-data-map { token-x: token-y, token-y: token-x }))
       )
       pair-already-exists-err
     )
+    (asserts!
+      (and
+        (is-eq (unwrap-panic (contract-call? .arkadiko-dao get-emergency-shutdown-activated)) false)
+        (is-eq (var-get swap-shutdown-activated) false)
+      )
```

Comment:
Extract this code into separate `read-only` function.
That way code will be a bit more readable and you'll have part of #320 already in place ;)